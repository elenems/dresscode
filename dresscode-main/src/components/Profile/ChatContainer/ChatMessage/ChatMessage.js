import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

function ChatMessage(props) {
  const style =
    props.userId === props.message.sender
      ? { background: "#f3f0f0" }
      : { background: "#fbfbfb" };
  return (
    <div style={style} className="chat-message">
      {props.userId === props.message.sender ? (
        <div className="chat-message-top">
          <p className="chat-message-messager">From you</p>
          <span>
            {new Date(props.message.postedAt).toString().slice(0, 24)}
          </span>
        </div>
      ) : (
        <div className="chat-message-top">
          <Link
            className="chat-message-messager"
            to={`/users/${props.message.sender}`}
          >
            {props.message.sender}
          </Link>
          <span>
            {new Date(props.message.postedAt).toString().slice(0, 24)}
          </span>
        </div>
      )}
      <div className="chat-message-container">
        <p className="chat-message-text">{props.message.messageText}</p>
      </div>
    </div>
  );
}

ChatMessage.propTypes = {
  userId: PropTypes.string.isRequired,
  message: PropTypes.object.isRequired
};

export default React.memo(ChatMessage);
