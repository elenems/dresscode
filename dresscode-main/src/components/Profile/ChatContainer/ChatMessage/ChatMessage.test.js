import { shallow } from "enzyme";
import React from "react";
import ChatMessage from "./ChatMessage";

describe("ChatMessage", () => {
  const props = {
    userId: "reveiver",
    message: {
      sender: "sender",
      postedAt: "10-10-2019",
      messageText: "Message text"
    }
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<ChatMessage {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Shows from you message if sender equals userId", () => {
    props.userId = "sender";
    const wrapper = shallow(<ChatMessage {...props} />);
    expect(wrapper.find(".chat-message-messager").text()).toBe("From you");
  });

  it("Shows message", () => {
    const wrapper = shallow(<ChatMessage {...props} />);
    expect(wrapper.find("p.chat-message-text").text()).toBe(
      props.message.messageText
    );
  });
});
