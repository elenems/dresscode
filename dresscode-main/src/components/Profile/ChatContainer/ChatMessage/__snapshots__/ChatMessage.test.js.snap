// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage Matches snapshot 1`] = `
"<div style={{...}} className=\\"chat-message\\">
  <div className=\\"chat-message-top\\">
    <Link className=\\"chat-message-messager\\" to=\\"/users/sender\\">
      sender
    </Link>
    <span>
      Thu Oct 10 2019 00:00:00
    </span>
  </div>
  <div className=\\"chat-message-container\\">
    <p className=\\"chat-message-text\\">
      Message text
    </p>
  </div>
</div>"
`;
